include/boost/geometry/algorithms/detail/overlay/overlay.hpp

```diff
@@ -389,15 +389,18 @@ std::cout << "traverse" << std::endl;
         // it can be returned or an exception can be thrown.
         return add_rings<GeometryOut>(selected_ring_properties, geometry1, geometry2, rings, out,
                                       strategy,
-                                      OverlayType == overlay_union ? 
 #if defined(BOOST_GEOMETRY_UNION_THROW_INVALID_OUTPUT_EXCEPTION)
+                                      OverlayType == overlay_union ?
                                       add_rings_throw_if_reversed
+                                      : add_rings_ignore_unordered
 #elif defined(BOOST_GEOMETRY_UNION_RETURN_INVALID)
+                                      OverlayType == overlay_union ?
                                       add_rings_add_unordered
+                                      : add_rings_ignore_unordered
 #else
                                       add_rings_ignore_unordered
 #endif
-                                      : add_rings_ignore_unordered);
+                                      );
```

Comment:
Thanks. Does anyone still remember why this only applies for union, and not for intersection or difference?

Reply:
I don't remember. It was added by me 5 years ago:
https://github.com/boostorg/geometry/commit/0b8a704ae95da0698e0c5f8b56c358e8e370a8ec
and then altered later:
https://github.com/boostorg/geometry/commit/fde547514442eb4d67aac78d7bf792ca373f6890

It seems it has something to do with spherical and geographic CS and the fact that if polygons are added together they may occupy more than half of the globe which would make holes from them because the area of such polygon would be negative. And other set operations cannot increase the size of the polygons in the result WRT the inputs.

Reply:
All right, thanks!